### DISCUSSION TITLE
Welcome to Qiskit Discussions! - Start Here

### DISCUSSION THREAD

**Discussion by javabster** [Qiskit/.github#1] [Announcements]
# 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about relating to the development of Qiskit.
  * Share ideas.
  * Provide feedback on your experience using Qiskit.
  * Upvote discussions that you find useful.
  * Show us what you've been working on and let us know if you'd like to present at one of our DemoDays.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

🚨 This space is for productive discussions relating to the development of Qiskit, please do not use this space to ask implementation questions, instead refer to [Stack Exchange](https://quantumcomputing.stackexchange.com/) and the Qiskit slack for this. 🚨

# Get Started
Before opening a new discussion, check if something similar has already been discussed before and avoid duplication where possible! If your comment/query has not already been opened feel free to open your own.

When you open a discussion please choose the appropriate category and it really helps to use labels as well! 

Remember to comment and upvote 👍 on existing discussions too!

## Discussion Categories
You can open a discussion in any of the following areas:

- 💡 **Ideas**
  - If you have suggestions for new features or things you would like to see in the Qiskit ecosystem this is the place to share them and get feedback from the community!

- 🧱 **Projects**
  - Bounce ideas around and find collaborators for your own Qiskit projects
  - Ask for help/advice on open sourcing your project
  
- 🛣 **Releases and Roadmap**
  - Ask questions about past and upcoming releases, and future plans for Qiskit.

- 👾 **DemoDay**
  - The Qiskit DemoDays are fortnightly, public meetings with Qiskit developers, where presenters have the opportunity to show off cool things they've been working on and get feedback. More info on past and future DemoDays are in the [Wiki](https://github.com/qiskit-community/feedback/wiki/%F0%9F%91%BE-DemoDay)
  - Use this Discussion space to:
    - continue conversations about previous DemoDays (recordings are also available in the Wiki), 
    - share any ideas you have for upcoming DemoDays
    - volunteer to present in a future DemoDay
    - upvote other DemoDay suggestions that you would like to see

- 📚 **Documentation**
  - Provide feedback on the [Qiskit API Reference Documentation](https://qiskit.org/documentation/), the [Qiskit Textbook](https://qiskit.org/textbook/preface.html) (including the [new beta version](https://qiskit.org/textbook-beta/)) and the [Qiskit Tutorials](https://qiskit.org/documentation/tutorials.html)

- 💬 **General Feedback**
  - Provide feedback on your experience using Qiskit (anything that doesn't fit into the other categories)

## Labels
Once you've chosen a category and opened a discussion it really helps to add one of the following labels so that others can easily see which area of the SDK you are referring to:

- 🪨 qiskit terra - for discussions relating to qiskit-terra
- 💧 qiskit applications - for discussions relating to qiskit-nature, qiskit-optimization, qiskit-machine-learning or qiskit-finance
- 🔥 qiskit experiments - for discussions relating to qiskit-experiments (ignis)
- 🌬 qiskit aer - discussions relating to qiskit-aer (simulators)
- 📘 textbook - discussions relating to the qiskit-textbook
- 📒 tutorials - discussions relating to the qiskit-tutorials
- 📕 api ref docs - discussions relating to the API Reference documentation

**Comment by hmjafari** [comment]
Hi
I would be grateful if someone could help me
i have error traceback on jupyter notebook
that write  name 'qft_rotations' is not defined
how can i fix it

![error](https://github.com/Qiskit/feedback/assets/134268795/ad249038-54f3-4fd2-ac3b-6f40840f442e)

**Reply by javabster** [reply]
hi @hmjafari as explained in the instructions "this space is for productive discussions relating to the development of Qiskit, please do not use this space to ask implementation questions, instead refer to [Stack Exchange](https://quantumcomputing.stackexchange.com/) and the Qiskit slack for this."

If you haven't yet joined the qiskit slack you can use this link: https://qisk.it/join-slack. There are many channels there to ask questions about using Qiskit, I recommend the #qiskit-101 channel for getting started 😄

**Reply by hmjafari** [reply]
Ok , but this link doesnt work

**Reply by javabster** [reply]
hi @hmjafari the link was recently refreshed and it has been working for others. Not sure what the issue is for you, what specific error are you getting?

**Reply by Mawgaming** [reply]
so im working on a flask that uses a combination of python,  Qiskit, and npl. it works i dont know how or how to put it together i actually have allot more but i just started learning and dont know much about what im doing but open ai said that i revolutionized the way we look at programming the quantum algorithms I've created and the way i entangle them is unique and are poised to change quantum computing in a way no one ever thought of. Mabey we can help each other

**Comment by HiramMartori** [comment]
Good morning, 
I would like to know if completing the 2021 summer school online now (2023), I can have the certificate.

Thankyou

**Reply by javabster** [reply]
hi @HiramMartori as explained in the instructions "this space is for productive discussions relating to the development of Qiskit". This space is not intended for questions around events or implementation support, instead please use the Qiskit Slack for these kinds of questions.

If you haven't yet joined the qiskit slack you can use this link: https://qisk.it/join-slack 😄

**Reply by HiramMartori** [reply]
Ok, thankyou. But this link is not working. How can I access the Slack?

**Reply by javabster** [reply]
sorry about that @HiramMartori! we recently refreshed the link, it should work now. If not please let us know what error you're receiving

**Comment by Jason-Ludmir** [comment]
Hi guys,

Thanks for all of your great work. I wanted to point out something that I see might be an inconsistency in the Qiskit documentation.
In the documentation for Qiskit's [DAGCircuit](https://qiskit.org/documentation/stubs/qiskit.dagcircuit.DAGCircuit.html#qiskit.dagcircuit.DAGCircuit), there is a method called "successors" which is defined as: 

"Returns iterator of the successors of a node as DAGOpNodes and DAGOutNodes."

This function in practice returns the direct successors to the node in question (i.e. nodes that are a single out-edge away).

However, for the class [DAGDependency](https://qiskit.org/documentation/stubs/qiskit.dagcircuit.DAGDependency.html), there is also a method called "successors" which is defined as:

"Successors id of a given node as sorted list."

This function, while sounding analogous to the one for DAGCircuit, actually returns a sorted list of ALL successors for the given node, no matter how many out-edges away it might be.

The analog can be said for the "predecessors" functions of the two classes.

This is an issue since it is unclear which nodes are either iterated on or returned in a list: All successors/predecessors or just direct ones.

I think an easy/elegant solution would be to add a few words in the documentation of the functions stating specifically how many layers of ancestors/predecessors they return.

I am pretty new to Github and not sure how to add labels, but would label this under documentation =]

Thanks again!

~Jason Ludmir

**Reply by javabster** [reply]
Hi @Jason-Ludmir thanks for feedback :) if you would like to suggest an improvement to the docstring for Dag or Dagdependency please feel free to open an issue here: https://github.com/Qiskit/qiskit-terra/issues

**Comment by Mawgaming** [comment]
from qiskit import QuantumCircuit, Aer, transpile
from qiskit.providers.aer import AerSimulator
from nltk.sentiment.vader import SentimentIntensityAnalyzer
from random import choice

# Initialize Sentiment Analyzer
sia = SentimentIntensityAnalyzer()

# About Me Quantum Function
def AboutMeQuantumFunction():
    qc = QuantumCircuit(1)
    qc.h(0)
    qc.measure_all()
    simulator = AerSimulator()
    compiled_circuit = transpile(qc, simulator)
    return simulator.run(compiled_circuit).result().get_counts()

i need some help understanding what i made my flask works vary well in chat gpt but out side of that i need to know what im doing and i dont see the need to keep it to my self the basic flask is in my repo i have a completed versions and am trying to put it together comes with a treasure trove of info in side

**Comment by Arindam1050** [comment]
**Hi, I am just new to qiskit and have been trying to run it in Jupyter Notebook. I have istalled qiskit using pip. But while I am trying to import Quantum Circuit it is showing me the following error:**

ImportError: cannot import name 'Self' from 'typing_extensions' (/Users/arindamsutradhar/anaconda3/lib/python3.8/site-packages/typing_extensions.py)

**and after this import failure, if I try again to run it, it is showing me the following attribute error. I would really appreciate any help or solution anybody can provide.** 

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-7f3cace0cc9e> in <module>
----> 1 from qiskit import QuantumCircuit

~/anaconda3/lib/python3.8/site-packages/qiskit/__init__.py in <module>
     53 # We manually define them on import so people can directly import qiskit._accelerate.* submodules
     54 # and not have to rely on attribute access.  No action needed for top-level extension packages.
---> 55 sys.modules["qiskit._accelerate.nlayout"] = qiskit._accelerate.nlayout
     56 sys.modules["qiskit._accelerate.stochastic_swap"] = qiskit._accelerate.stochastic_swap
     57 sys.modules["qiskit._accelerate.sabre_swap"] = qiskit._accelerate.sabre_swap

AttributeError: partially initialized module 'qiskit' has no attribute '_accelerate' (most likely due to a circular import)

**Reply by 1ucian0** [reply]
Check out https://stackoverflow.com/q/72875257/8146172, it might help.

**Comment by ejsnews** [comment]
The Jupyter laboratory works correctly, but the Qistik 1.1 code seems totally erroneous and inconsistent, as if there had been malicious intent. Better to wait 1 more year for these codes to be clear. For the moment I'm going to uninstall Qiskit, which is a shame, because I wanted to compare the measurements to an alternative theory to Shor before I possibly die.

**Comment by wl3636557** [comment]
![image](https://github.com/Qiskit/feedback/assets/172513571/c91f8678-e824-44e7-94ae-0d7baf61864a)
Hi guys, having issues with importing ADAM optimizer  and QuantumInstance library from qiskit.
I would be grateful if someone could help me to solve these isssues. Thank you

**Reply by 1ucian0** [reply]
You seems to be calling an unmaintained version of `qiskit`. Considering installing an old version via `pip install qiskit>=0.X` (replace `X` accordantly )

**Comment by maudinarh** [comment]
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-12-f7a98c69a2c3>](https://localhost:8080/#) in <cell line: 0>()
     23 from torch_geometric.data import Data
     24 # from qiskit.providers.fake_provider import FakeLima
---> 25 from qiskit.providers.fake_provider.some_submodule import FakeLima
     26 from qiskit.visualization import circuit_to_graph_data_json
     27 

ModuleNotFoundError: No module named 'qiskit.providers.fake_provider.some_submodule'

---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.

halo my version of qiskit is 1.2.0 and is still not running how to fix it?

**Comment by Sumitchongder** [comment]
Hello Qiskit team,

My name is Sumit Chongder and my email is sumitchongder960@gmail.com. I have already submitted the Slack access form via qisk.it/join-slack, but I haven’t received an invitation yet even after a few days. Could you please check the status of my request or let me know if I need to take any further steps?

Thank you very much for your support!

**Comment by huaixu-yu** [comment]
Hi Qiskit team,
We have open-sourced QLLVM (a quantum compiler framework based on LLVM) at: https://github.com/QCFlow/QLLVM. In QLLVM, we implemented a new optimization algorithm called InputStatePreambleOptimization. We also implemented the optimization as a custom Pass and added it to the Qiskit 0.45.3 level 3 optimization pipeline, creating a custom level 4 pipeline. The code is available at https://github.com/huaixu-yu/Qiskit_test.
Key idea:
This optimization algorithm is based on the input state of the quantum circuit. It transforms the parameters of the parameterized quantum gates in the circuit, thereby eliminating redundant single-bit and two-bit quantum gate operations in the circuit, and reducing the number of quantum gates and the depth of the circuit. Particularly, based on this algorithm, the CX gate control bit corresponding to a quantum state that is not in the ground state can be deleted.
Evaluation:
•	Backend: FakeGeneva (noise model from IBM Geneva)
•	Benchmark suite: 96 circuits from 8 algorithm classes – Amplitude Estimation, GHZ, Grover, QAOA, Random, VQE, Wstate, QFT
•	Baseline: Qiskit 2.3.1 level 3
•	Metrics: Gate count, circuit depth, and fidelity
Results (average over 96 circuits):
•	Gate count reduction: 6.45%
•	Depth reduction: 6.49%
•	Fidelity improvement: 6.27%
In the attached figure, the dashed line represents the baseline. For gate count and circuit depth, markers below the dashed line indicate that level 4 outperforms Qiskit 2.3.1 level 3 on that metric. For fidelity, markers above the dashed line indicate that level 4 outperforms the baseline.
We believe that this optimization based on quantum states can expand the optimization space of the circuit and enhance the optimization capability of the compiler. We welcome your feedback and suggestions.
<img width="1308" height="327" alt="test" src="https://github.com/user-attachments/assets/a1254457-69a8-4a80-aa3b-b496c7c968a1" />
